### ISSUE TITLE
Race condition between between server creation and host shutdown

### ISSUE THREAD

**Issue by Cooldude2606** [clusterio/clusterio#676]
TL;DR There is a breif period where server state is "created" but the processes has not been spawned, this creates a race condition during shutdown of a host.

When a host is started with auto starting instances, they will immediately start their servers; but if the host is also shutdown at the same time, such as during CI tests, then there is a brief period where server state is "created" but the server process is not yet spawned. This occours because of the time needed to write the server config and map gen settings to the disk, a large latency or slow write increases the chance of a crash and an orphan server process.

The code below is where the issues occurs: (within `@host/server.ts::create`)
```ts
this._state = "create";
await this._writeConfigIni();
await this._writeMapSettings(mapGenSettings, mapSettings);
this._server = child_process.spawn( ... )
```

The suggested fix is to make these into synchronous writes to remove the option for shutdown / kill to be called during the transient state. Error handling will need to be performed to return to a valid state if the writes / spawn throw an error.
